File: RelacjeSportowe.DataAccess/Dtos/Requests/TransmissionRequest.cs
using System;

namespace RelacjeSportowe.DataAccess.Dtos.Requests
{
    public abstract class TransmissionRequest
    {
        public string TeamHome { get; set; }

        public string TeamAway { get; set; }

        public DateTime EventDate { get; set; }
    }

    public class AddTransmissionRequest : TransmissionRequest
    {
    }

    public class DeleteTransmissionRequest
    {
        public int Id { get; set; }
    }

    public class EndTransmissionRequest
    {
        public int Id { get; set; }
    }

    public class UpdateTransmissionRequest : TransmissionRequest
    {
        public int Id { get; set; }
    }

    public class VoteTransmissionRequest
    {
        public int Id { get; set; }

        public int Rating { get; set; }
    }
}

File: RelacjeSportowe.Web/ClientApp/src/app/models/dtos/requests/transmission-request.ts
abstract class TransmissionRequest {
  public teamHome: string;
  public teamAway: string;
  public eventDate: Date;

  constructor(data: any) {
    this.teamHome = data.teamHome;
    this.teamAway = data.teamAway;

    if (data.eventDate && data.eventTime) {
      this.eventDate = new Date(`${data.eventDate}T${data.eventTime}:00`);
    } else {
      this.eventDate = data.eventDate;
    }
  }
}

export class AddTransmissionRequest extends TransmissionRequest {
  constructor(data: any) {
    super(data);
  }
}

export class DeleteTransmissionRequest {
  public id: number;

  constructor(data: any) {
    this.id = data.id;
  }
}

export class EndTransmissionRequest {
  public id: number;

  constructor(data: any) {
    this.id = data.id;
  }
}

export class UpdateTransmissionRequest extends TransmissionRequest {
  public id: number;

  constructor(data: any) {
    super(data);
    this.id = data.id;
  }
}

export class VoteTransmissionRequest {
  public id: number;
  public rating: number;

  constructor(data: any) {
    this.id = data.id;
    this.rating = data.rating;
  }
}